### PR TITLE
Change Node version validator to use 'nvm current'

### PR DESCRIPTION
**Pull request by davprat**
### Description of changes
* Use `nvm current` to get the current version of Node.js if `node --version` fails
* Raise an exception if `nvm current' returns a major version is greater than a max major version asking the user to install the latest release of 'max major version'.

### Tests
* Added test to make sure that the fallback to `nvm current` succeeds if the call to `node --version` fails.

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/api/util.py [hunk 1]
```diff
@@ -177,19 +177,46 @@ def _assert_node_executable():
...
+        except Exception:
+            message = "Unable to check Node.js version"
+            LOGGER.critical(message)
+            raise Exception(message)
+        if node_version.major > packaging.version.parse(NODEJS_MAX_MAJOR_VERSION).major:
+            message = (
+                f"Node.js version {node_version_string} may not work on this platform. Use the Node"
+                f" Version Manager (nvm) to install and use the latest release of Node.js version"
+                f" {NODEJS_MAX_MAJOR_VERSION}."
+            )
```
Comment (demartinofra) [inline cli/src/pcluster/api/util.py]: I'd prefer if possible not to hardcode NODEJS_MAX_MAJOR_VERSION in our CLI. Can we just suggest to make sure node is properly installed and functional if we detect that node is available but returning some errors?

Comment (davprat) [reply cli/src/pcluster/api/util.py]: Removed and reworked.

File: cli/src/pcluster/api/util.py [hunk 2]
```diff
@@ -177,19 +177,47 @@ def _assert_node_executable():
...
+            #   untrusted source
+            node_version_string = subprocess.check_output(  # nosec
+                ["nvm", "current"], stderr=subprocess.STDOUT, shell=False, encoding="utf-8"
+            )
+            LOGGER.debug("Found Node.js version '%s' in use", node_version_string)
+        except Exception:
+            message = "Unable to check Node.js version"
+            LOGGER.critical(message)
+            raise Exception(message)
+        if node_version_string == "none":
```
Comment (demartinofra) [inline cli/src/pcluster/api/util.py:205]: [optional] might be useful to tell in a comment what command we expect to return the "none" string. is it `nvm current`? do we need a test case for this?

**Review by demartinofra — DISMISSED** [review]
LGTM, left a minor comment

File: cli/src/pcluster/api/util.py (hunk 2 above)
Comment (davprat) [reply cli/src/pcluster/api/util.py:205]: I think this [test case](https://github.com/aws/aws-parallelcluster/pull/4493/files#diff-9350f46969010c1d42950e607a45080fda852e9c43b1939bd821b40212586734R59) covers the none response? Let me know if I'm missing something though.

Comment (davprat) [reply cli/src/pcluster/api/util.py:205]: Added comment.

**Review by demartinofra — APPROVED** [review]

**Pull request merged**